packages/core/src/actors.ts

```diff
@@ -366,63 +366,71 @@ export function fromMachine<TMachine extends AnyStateMachine>(
   > extends true
     ? TMachine
     : 'Some implementations missing',
-  options?: Partial<InterpreterOptions>
+  options: Partial<InterpreterOptions> = {}
 ): Behavior<EventFrom<TMachine>, StateFrom<TMachine>> {
+  const snapshotEventType = Symbol('snapshot');
+
   const castedMachine = machine as TMachine;
   let service: InterpreterFrom<TMachine> | undefined;
-  let subscription: Subscription;
   let initialState: StateFrom<TMachine>;
 
-  const behavior: Behavior<EventFrom<TMachine>, StateFrom<TMachine>> = {
+  // TODO: use better type for `TEvent`
```

Comment:
The problem is that we are also handling signals here, so we cant simply use `TEvent` here. We also can't use smth like `TEvent | LifecycleEvent` because we also need to process `typeof snapshotEventType`.

I'm not yet sure what exact design to choose here - this works right now but I wonder if there is something cleaner~ out there. I think that it would be cool to ditch usage of symbols cause they don't give us that much and they are breaking the compatibility with the `EventObject` constraint (after all a symbol doesn't extend a string). At the same time we don't want those events/signals to "clash" with whatever the user can define for their `TEvent`

Reply:
In the future, we can improve the `Behavior` interface:

```ts
interface Behavior {
  transition: (snapshot, event, ctx) => snapshot;
  onStart?: () => snapshot;
  onStop?: () => snapshot;
  // ...
}
```

And put the start and stop behavior in separate callbacks, similar to `Behaviors.setup(fn)` in Akka (see example here: https://doc.akka.io/docs/akka/current/typed/actors.html)

Reply:
Ye, those could be perhaps handled like this - but the described problem here wouldn't totally go away with such an approach. Here, we are also handling some other, "internal", events - like `snapshotEventType`. This doesn't quite fit the idea of signals, it's not related to lifecycle etc.

In theory, we could wrap regular events (`TEvent`) in another custom event - call `transition` with it, handle that in the switch case, and "forward" the unpacked even in the dedicated case. This was the type of this `behavior` would become smth like `Behavior<LifecycleEvent | typeof snapshotEventType | EventWrapper<TEvent>, StateFrom<TMachine>>`. 

It would help us out with the types and we wouldn't be able to end up with any namespace clashes. OTOH it feels like juggling stuff a little bit too much, OYAH programming is about juggling stuff :sweat_smile: